core/src/main/scala/org/allenai/common/ParIterator.scala

```diff
@@ -0,0 +1,58 @@
+package org.allenai.common
+
+import java.util.concurrent.Semaphore
+
+import scala.concurrent._
+import scala.concurrent.duration.Duration
+
+object ParIterator {
+  val queueLimit = 1024
```

Comment:
I think we should make this a configurable parameter.  Otherwise the client has no recourse if they want to use this library and performance tune.